File: C++/Medium/Accepted/2-keys-keyboard_6.cpp
class Solution {
public:
    int minSteps(int n) {
        
        int i = 2;
        int count = 0;
        while(n > 1){
            while(n % i == 0){
                count += i;
                n /= i;
            }
            i++;
        }
        return count;
    }
};

File: Rust/Medium/CompileError/2-keys-keyboard_6.rs
impl Solution {
    pub fn min_steps(n: i32) -> i32 {
        

        let n = n as usize;
        if n == 1 {
            return 0;
        }
        let mut dp = vec![0; n + 1];
        for i in 2..=n {
            dp[i] = i;
            for j in 2..i {
                if i % j == 0 {
                    dp[i] = dp[j] + i / j;
                    break;
                }
            }
        }
        dp[n]
    }
}